### PR TITLE
refactor(sink): reimplement remote sink without writer

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

As titled. This is a preliminary PR for async log truncation for jdbc sink.

Besides, the generic type `R: RemoteSinkTrait` is removed from RemoteLogSinker and RemoteCoordinator.

The `RemoteSinkWriter` is removed and the generic `SinkWriterInner` is only used by `CoordinatedRemoteSinkWriter`, and therefore in this PR we remove the generic parameter on `SinkWriterInner` and it is renamed to `CoordinatedRemoteSinkWriter`.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/JniSinkWriterResponseObserver.java [hunk 1]
```diff
@@ -25,22 +26,24 @@ public class JniSinkWriterResponseObserver
...
-                Binding.sendSinkWriterResponseToChannel(this.responseTxPtr, response.toByteArray());
+        if (!Binding.sendSinkWriterResponseToChannel(this.responseTxPtr, response.toByteArray())) {
+            throw Status.INTERNAL.withDescription("unable to send response").asRuntimeException();
+        }
     }
 
     @Override
     public void onError(Throwable throwable) {
         this.success =
                 Binding.sendSinkWriterErrorToChannel(this.responseTxPtr, throwable.getMessage());
```
Comment (hzxa21) [inline java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/JniSinkWriterResponseObserver.java:45]: nits: we don't need to assign `this.success` here.

Also, shall we throw exception here if the channel is close to make it consistent with `onNext`?

File: src/connector/src/sink/remote.rs [hunk 2]
```diff
@@ -411,145 +402,55 @@ impl<SM, R: RemoteSinkTrait> RemoteSinkWriterInner<SM, R> {
...
             epoch: None,
             batch_id: 0,
             stream_handle,
-            payload_format: connector_params.sink_payload_format,
-            json_encoder: JsonEncoder::new(
-                schema,
-                None,
-                TimestampHandlingMode::String,
-                TimestamptzHandlingMode::UtcWithoutSuffix,
-            ),
```
Comment (hzxa21) [inline src/connector/src/sink/remote.rs:427]: Will we configure the json encoder elsewhere?

**Review by hzxa21 — APPROVED** [review]
Rest LGTM

File: java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/JniSinkWriterResponseObserver.java (hunk 1 above)
Comment (wenym1) [reply java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/JniSinkWriterResponseObserver.java]: fixed.

In `onError` setting the success flag shall be enough. The handler will stop when it detects that the flag is set false. I have added a warning log to print the unsent error message.

File: src/connector/src/sink/remote.rs (hunk 2 above)
Comment (wenym1) [reply src/connector/src/sink/remote.rs:427]: No. The json format for payload is deprecated in this PR.

**Pull request merged**